service-map/src/main/java/fi/mml/map/mapwindow/util/OskariLayerWorker.java

```diff
@@ -125,16 +150,35 @@ public static JSONObject getListOfMapLayers(final List<OskariLayer> layers, fina
         final JSONArray layersList = new JSONArray();
         start = System.currentTimeMillis();
         for (OskariLayer layer : layers) {
+        	
             final String permissionKey = layer.getType()+ "+" + layer.getUrl() + "+" + layer.getName();
             if (layer.getParentId() == -1 && !resources.contains(permissionKey)) {
                 // not permitted if resource NOT found in permissions!
                 // sublayers can pass through since their parentId != -1
                 continue;
             }
             try {
+            	String boundingBoxWKT = layer.getCapabilities().optString("geom");
```

Comment:
These bounding box changes seem a bit out of place inside this PR. There seems to be no notes of changing any of this in the PR comment or in the change log. What's the reasoning for this changes? Are they necessary for the Hierarhical layerlist somehow?

Reply:
Yes it is outside this PR. But also one requirement when creating hierarchical layerlist is at layer tools can zoom to layer extent. I add description to PR.

Reply:
This doesn't seem to work. There's a transformWKTGeom() method available in this class that already does this thing correctly. You can remove this code and add a call to transformWKTGeom() below.